lite/backends/nnadapter/nnadapter/driver/huawei_ascend_npu/utility.cc

```diff
@@ -492,5 +519,105 @@ std::string ConvertInterpolateModeCodeToGEInterpolateMode(
+        ascend_cann_path = path;
+        break;
+      }
+    }
+    if (ascend_cann_path.empty()) {
+      ascend_cann_path = "/usr/local/Ascend/ascend-toolkit/latest";
+      NNADAPTER_LOG(ERROR) << "Unable to find the Ascend CANN installation "
+                              "path, use the default path["
+                           << ascend_cann_path << "]";
+    }
+    auto ascend_cann_real_path = GetRealPath(ascend_cann_path.c_str());
+    std::vector<std::string> ascend_cann_real_path_split =
+        string_split<std::string>(ascend_cann_real_path, "/");
+    std::string cann_version;
+    for (size_t i = 0; i < ascend_cann_real_path_split.size(); i++) {
+      if (ascend_cann_real_path_split[i] == "ascend-toolkit" ||
+          ascend_cann_real_path_split[i] == "nnrt") {
+        if (i <= ascend_cann_real_path_split.size() - 1) {
+          cann_version = ascend_cann_real_path_split[i + 1];
+          break;
+        } else {
+          NNADAPTER_LOG(ERROR) << "Unable to find the version of Ascend CANN";
+          return std::make_tuple(0, 0, 0, 0);
+        }
+      }
+    }
+    auto cann_version_split = string_split<std::string>(cann_version, ".");
+    if (cann_version_split.size() == 3) {
+      major_version = atoi(cann_version_split[0].c_str());
+      minor_version = atoi(cann_version_split[1].c_str());
+      patch_version = atoi(cann_version_split[2].c_str());
+    } else if (cann_version_split.size() == 4) {
+      major_version = atoi(cann_version_split[0].c_str());
+      minor_version = atoi(cann_version_split[1].c_str());
+      patch_version = atoi(cann_version_split[2].c_str());
+      std::regex reg("(\\d+)");
+      std::smatch match_result;
+      if (std::regex_search(cann_version_split[3], match_result, reg)) {
+        if (match_result.size() >= 1) {
+          bugfix_version = atoi(match_result[0].str().c_str());
+        } else {
+          bugfix_version = 0;
+        }
+      }
+    } else {
+      NNADAPTER_LOG(ERROR) << "Unable to get the version of Ascend CANN";
+      return std::make_tuple(0, 0, 0, 0);
+    }
+    initialized = true;
+  }
+  mtx.unlock();
+  return std::make_tuple(
+      major_version, minor_version, patch_version, bugfix_version);
+}
+
+std::string AscendCANNVersion2String(int major_version,
+                                     int minor_version,
+                                     int patch_version,
+                                     int bugfix_version) {
+  if (bugfix_version == 0)
+    return string_format(
+        "%d.%d.%d", major_version, minor_version, patch_version);
+  else
+    return string_format("%d.%d.%d.alpha%d",
+                         major_version,
+                         minor_version,
+                         patch_version,
+                         bugfix_version);
+}
+
+ge::AscendString GetAscendSocName() {
+  ge::AscendString soc_version = "Ascend310";
+#if NNADAPTER_HUAWEI_ASCEND_NPU_CANN_MINIMUM_SUPPORTED_VERSION(5, 0, 2, 1)
```

Comment:
#if NNADAPTER_HUAWEI_ASCEND_NPU_CANN_MINIMUM_SUPPORTED_VERSION(5, 0, 2, 1)
  const char* soc_name = aclrtGetSocName();
  if (soc_name != nullptr) {
    soc_version = ge::AscendString(soc_name);
  } else {
    NNADAPTER_LOG(WARNING)
        << "Failed to call  aclrtGetSocName  to obtain the SoC name, so Ascend 310 is used by default.";
  }
#else
  NNADAPTER_LOG(WARNING) << "Since the current library is compiled based on CANN versions below 5.0.2.1, aclrtGetSocName cannot be called to obtain the SoC name of the current device, so Ascend 310 is used by default. If you want to use ascend 710, please recompile the library based on CANN 5.0.2.1 and above.";
#endif

Reply:
done